lib/ansible/modules/network/dnac/dnac_rest.py

```diff
@@ -0,0 +1,236 @@
+
+    # if the user is working with this module in only check mode we do not
+    # want to make any changes to the environment, just return the current
+    # state with no modifiupcations
+    if module.check_mode:
+        return result
+
+    # manipulate or modify the state as needed (this is going to be the
+    # part where your module will do what it needs to do)
+    protocol = None
+    if module.params['use_ssl'] is not None and module.params['use_ssl'] is False:
+        protocol = 'http'
+    else:
+        protocol = 'https'
+
+    url = '{0}://{1}{2}'.format(protocol, module.params['host'], module.params['path'])
+    headers = {'Content-Type': 'application/json'}
+    authheaders = {'Content-Type': 'application/json'}
+
+    try:
+        authurl = '{0}://{1}/api/system/v1/auth/login'.format(protocol, module.params['host'])
+        authresp = open_url(authurl,
+                            headers=authheaders,
+                            method='GET',
+                            use_proxy=module.params['use_proxy'],
+                            timeout=module.params['timeout'],
+                            validate_certs=module.params['validate_certs'],
+                            url_username=module.params['username'],
+                            url_password=module.params['password'],
+                            force_basic_auth=True,
+                            )
+
+    except Exception as e:
+        module.fail_json(msg=e)
+
+    if to_native(authresp.read()) != "success":  # DNA Center returns success in body
+        module.fail_json(msg="Authentication failed: {0}".format(authresp.read()))
+
+    respheaders = authresp.getheaders()
+    cookie = None
+    for i in respheaders:
+        if i[0] == 'Set-Cookie':
+            cookie_split = i[1].split(';')
+            cookie = cookie_split[0]
+
+    if cookie is None:
+        module.fail_json(msg="Cookie not assigned from DNA Central")
+
+    headers['Cookie'] = cookie
+
+    try:
+        resp, info = fetch_url(module, url,
+                               data=payload,
+                               headers=headers,
+                               method=module.params['method'].upper(),
+                               use_proxy=module.params['use_proxy'],
+                               force=True,
+                               timeout=module.params['timeout'],)
+
+    except Exception:
+        module.fail_json(msg=resp['status'])
+
+    if info['status'] != 200:
+        module.fail_json(msg='{0}: {1} '.format(info['status'], info['body']))
+
+    # use whatever logic you need to determine whether or not this module
+    # made any modifications to your target
+    if info['status'] == 200:
+        result['changed'] = True
+        # result['message']=json.loads(resp.read())
+        try:
+            result['message'] = json.loads(resp.read())
+        except:
+            module.fail_json(msg="DNA Center didn't return JSON compatible data")
```

Comment:
I think it's better to refrain from abbreviations in formal/written text. (At least that's what we learned at school)
So make that:
```python
    module.fail_json(msg="DNA Center did not return JSON compatible data")`
```
So it doesn't look like nested quotes.